monitors/SyntheticPegMonitor.js

```diff
@@ -0,0 +1,101 @@
+const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+const { createObjectFromDefaultProps } = require("../common/ObjectUtils");
+
+class SyntheticPegMonitor {
+  /**
+   * @notice Constructs new Liquidator bot.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} web3 Instance of a web3 client provided by the class that initiates the monitor module.
+   * @param {Object} uniswapPriceFeed Module used to query the current uniswap token price.
+   * @param {Object} medianizerPriceFeed Module used to query the current crypto watch token price.
+   * @param {Object} [config] Contains fields with which constructor will attempt to override defaults.
+   */
+  constructor(logger, web3, uniswapPriceFeed, medianizerPriceFeed, config) {
+    this.logger = logger;
+
+    // Instance of price feeds used to check for deviation of synthetic token price.
+    this.uniswapPriceFeed = uniswapPriceFeed;
+    this.medianizerPriceFeed = medianizerPriceFeed;
+
+    this.web3 = web3;
+
+    // Contract constants
+    // TODO: replace this with an actual query to the collateral currency symbol
+    this.collateralCurrencySymbol = "DAI";
+    this.syntheticCurrencySymbol = "UMATEST";
+
+    // TODO: get the decimals of the collateral currency and use this to scale the output appropriately for non 1e18 colat
+    this.formatDecimalString = createFormatFunction(this.web3, 2);
+
+    // Default config settings. SyntheticPegMonitor deployer can override these settings by passing in new
+    // values via the `config` input object. The `isValid` property is a function that should be called
+    // before resetting any config settings. `isValid` must return a Boolean.
+    const { toBN, toWei } = this.web3.utils;
+    const defaultConfig = {
+      deviationAlertThreshold: {
+        // `deviationAlertThreshold`: Error threshold used to compare observed and expected token prices.
+        // if the deviation in token price exceeds this value an alert is fired.
+        value: this.web3.utils.toBN(this.web3.utils.toWei("0.2")),
+        isValid: x => {
+          return toBN(x).lte(toBN(toWei("100"))) && toBN(x).gte(toBN("0"));
+        }
+      }
+    };
+    Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));
+  }
+
+  // Queries disputable liquidations and disputes any that were incorrectly liquidated.
+  checkPriceDeviation = async () => {
+    this.logger.debug({
+      at: "SyntheticPegMonitor",
+      message: "Checking price deviation"
+    });
+    // Get the latest prices from the two price feeds.
```

Comment:
Both prices feeds integration here use `getCurrentPrice`. Do we want to consider using a short twap? it feels most intuitive to me to just get the current price but am happy to change this.

Reply:
We could do a short twap -- we could also draft up a normal uniswap price feed that doesn't do a twap. Not a high priority, but wouldn't take much time. I do think a twap of 1 second is a little janky, but it basically resolves to the current price IIRC.

Regardless, I think using `getCurrentPrice` on both allows this monitor to generalize such that it could be used to monitor discrepancies between any two price feeds. For instance, we could use it to monitor deviations between exchanges as a measure of market volatility or something.